// Yandex ASR dictation api (draft):
// Client initiate session with http upgrade request, for example:
// GET /asr_partial HTTP/1.1\r\n
// User-Agent:KeepAliveClient\r\n
// Host: voice-stream.voicetech.yandex.net:80\r\n
// Upgrade: dictation\r\n\r\n
// Receive HTTP 101 Switched response.
// Next send\receive protobuf messages, format
// [hex size]\r\n[message body serialized with protobuf]

// send ConnectionRequest, read ConnectionsResponse.. etc
// send AddData, read AddDataResponse and so on.
syntax = "proto2";

package BasicProtobuf;

message ConnectionResponse
{
  required ResponseCode responseCode = 1;

  required string sessionId = 2;

  optional string message = 3;

  enum ResponseCode {
    OK = 200;
    BadMessageFormatting = 400;
    UnknownService = 404;
    NotSupportedVersion = 405;
    Timeout = 408;
    ProtocolError = 410;
    InternalError = 500;
    InvalidKey = 429;
    InvalidRequestParams = 406;
	}
}

